boards/FSM/Test/Src/Test_StateMachine.cpp

```diff
@@ -110,3 +130,27 @@ TEST_F(FsmStateMachineTest, check_air_closed_state_is_broadcasted_over_can)
         CANMSGS_FSM_STATE_MACHINE_STATE_AIR_CLOSED_CHOICE,
         App_CanTx_GetPeriodicSignal_STATE(can_tx_interface));
 }
+
+TEST_F(
+    FsmStateMachineTest,
+    check_if_primary_and_secondary_flow_rates_are_broadcasted_over_can_in_all_states)
+{
+    float fake_frequency = 1.0f;
+
+    for (const auto &state : GetAllStates())
+    {
+        SetInitialState(state);
+        get_primary_flow_rate_fake.return_val   = fake_frequency;
+        get_secondary_flow_rate_fake.return_val = fake_frequency;
+        App_SharedStateMachine_Tick(state_machine);
+
+        EXPECT_EQ(
+            fake_frequency,
+            App_CanTx_GetPeriodicSignal_PRIMARY_FLOW_RATE(can_tx_interface));
+        EXPECT_EQ(
+            fake_frequency,
+            App_CanTx_GetPeriodicSignal_SECONDARY_FLOW_RATE(can_tx_interface));
+
+        // To avoid false positives, we use a different frequency each time
+    }
```

Comment:
You removed fake_frequency++?  :thought_balloon: 

```suggestion
        fake_frequency++;
    }
```

Reply:
This may have been a 3AM moment